src/db_impl.cc

```diff
@@ -837,6 +837,46 @@ Status TitanDBImpl::DeleteFilesInRanges(ColumnFamilyHandle* column_family,
   return s;
 }
 
+int TitanDBImpl::CountSortedRuns(
+    const std::vector<std::shared_ptr<BlobFileMeta>>& files) {
+  std::list<BlobFileMeta*> remained_files;
+  std::vector<int> sorted_run_blobs;
+  int num_files = files.size();
+
+  auto blob_cmp = [](const BlobFileMeta* f1, const BlobFileMeta* f2) {
+    return f1->smallest_key().compare(f2->smallest_key()) < 0;
+  };
+  for (const auto& file : files) {
+    remained_files.emplace_back(file.get());
+  }
+  remained_files.sort(blob_cmp);
+
+  while (!remained_files.empty()) {
+    int cnt = 0;
+    BlobFileMeta* pre = nullptr;
+    for (auto iter = remained_files.begin(); iter != remained_files.end();) {
+      if (pre == nullptr ||
+          (*iter)->smallest_key().compare(pre->largest_key()) > 0) {
+        cnt++;
+        pre = *iter;
+        remained_files.erase(iter++);
+      } else {
+        iter++;
+      }
+    }
+    sorted_run_blobs.push_back(cnt);
+  }
+
+  // There might be several sorted runs consist of few wide range blob files.
+  // Ignore these sorted runs since it has almost no impact to scan performance.
+  std::sort(sorted_run_blobs.begin(), sorted_run_blobs.end());
+  int i = 0, ignore = num_files * 0.1;
+  while (sorted_run_blobs[i] < ignore) {
+    ignore -= sorted_run_blobs[i++];
```

Comment:
why `ignore` should minus the ignored runs?

Reply:
sorted_run_blobs is number of blob files in each sorted run, we ignore 10% files since some sorted runs have only very few files which has no much impacts to scan performance.

Reply:
oh, I misunderstand the code.